### PR TITLE
server: support partitioned table for the /mvcc/* HTTP API

### PR DESCRIPTION
**Pull request by tiancaiamao**
<!--
Thank you for contributing to TiDB! Please read TiDB's [CONTRIBUTING](https://github.com/pingcap/community/blob/master/CONTRIBUTING.md) document **BEFORE** filing this PR.
-->

### What problem does this PR solve? <!--add issue link with summary if exists-->

```
mysql> show create table xx;
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                                                                                                                                                                                                     |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| xx    | CREATE TABLE `xx` (
  `id` int(11) DEFAULT NULL,
  `ts` timestamp NULL DEFAULT NULL,
  KEY `i_ts` (`ts`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin
PARTITION BY RANGE ( `id` ) (
  PARTITION p0 VALUES LESS THAN (1),
  PARTITION p1 VALUES LESS THAN (100),
  PARTITION p2 VALUES LESS THAN (1000),
  PARTITION pn VALUES LESS THAN (10000)
) |
+-------+------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

mysql> select * from xx;
+------+---------------------+
| id   | ts                  |
+------+---------------------+
|    1 | 2019-12-30 12:23:33 |
+------+---------------------+
1 row in set (0.01 sec)
```

Use the `table(partition)` name pattern to work on the partitioned table.

```
➜  tidb git:(mvcc-http-api) curl 'http://127.0.0.1:10080/mvcc/key/test/xx(p1)/1'  
{
 "key": "7480000000000000515F728000000000000001",
 "region_id": 1017,
 "value": {
  "info": {
   "writes": [
    {
     "start_ts": 413431571564199937,
     "commit_ts": 413431571577307137,
     "short_value": "CAIIAggECYCAgIjeiL/SGQ=="
    }
   ]
  }
 }
}
```

```
➜  tidb git:(mvcc-http-api) curl http://127.0.0.1:10080/mvcc/index/test/xx\(p1\)/i_ts/1\?ts\=2019-12-30%2004:23:33 
{
 "key": "7480000000000000515F6980000000000000010419A4FC45E1000000038000000000000001",
 "region_id": 1017,
 "value": {
  "info": {
   "writes": [
    {
     "start_ts": 413431571564199937,
     "commit_ts": 413431571577307137,
     "short_value": "MA=="
    }
   ]
  }
 }
}
```

### What is changed and how it works?


### Check List <!--REMOVE the items that are not applicable-->

Tests <!-- At least one of them must be included. -->

 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to update the documentation

### CODE REVIEW

**Comment by Deardrops** [comment]
how about make table partition number a http parameter?
http://127.0.0.1:10080/mvcc/key/test/xx/1?partition=1

**Comment by Deardrops** [comment]
By the way, can we query more than 1 partition or all partition in one http query?

**Comment by tiancaiamao** [comment]
> how about make table partition number a http parameter?
> http://127.0.0.1:10080/mvcc/key/test/xx/1?partition=1

I've thought about this method before, but it has a drawback.  @Deardrops 
There is an API like this:

```
/mvcc/index/{db}/{table}/{index}/{handle}?col1=xxx&col2=xxx&co3=xxx
```

If we use `?partition=1` we can't distinguish whether it's a column named 'partition', or we want to query on a partitioned table, so I prefer `table(partition)`

**Comment by tiancaiamao** [comment]
> By the way, can we query more than 1 partition or all partition in one http query?

This API is used for debugging, I don't see any real needs to query multiple partitions in practice.

**Comment by tiancaiamao** [comment]
PTAL @Deardrops @zimulala

File: docs/tidb_http_api.md [hunk 1]
```diff
@@ -252,6 +252,12 @@ timezone.*
     }
     ```
 
+    *Hint: On a partitioned table, use the `table(partition)` pattern as the table name, `test(p1)` for example:
```
Comment (jackysp) [inline docs/tidb_http_api.md]:
```suggestion
    *Hint: On a partitioned table, use the `table(partition)` pattern as the table name, `test(p1)` for example:*
```

**Comment by tiancaiamao** [comment]
PTAL @jackysp @zimulala

**Comment by tiancaiamao** [comment]
PTAL @Deardrops

**Comment by Deardrops** [comment]
LGTM

**Pull request merged**